File: src/main/adapter-trait/java/multiinheritance/Pegase.java
package multiinheritance;

/*
This tricky code is outdated using Java 8 and traits (default interfaces).
 */
public class Pegase extends Horse implements Fly {
	// Looks like a Proxy but its an Adapter: the goal is two bind incompatible objects (Pegase with Bird)
	private final Bird bird = new Bird();

	@Override
	public void fly() {
		bird.fly();
	}
}

File: src/main/adapter-trait/go/insa/pegase.go
package main

import "fmt"

type Bird struct {
}

func (bird Bird) fly() {
   fmt.Println("fly")
}

type Horse struct {
}

func (horse Horse) gallop() {
    fmt.Println("gallop")
}

type Pegase struct {
// This is not inheritance but composition
  Horse
  Bird
}


func main() {
    var pegase Pegase

    pegase.gallop()
    pegase.fly()
}